app/models/miq_database.rb

```diff
@@ -87,4 +87,57 @@ def verify_credentials(auth_type = nil, _options = {})
   def registration_organization_name
     registration_organization_display_name || registration_organization
   end
+
+  def ansible_secret_key
+    auth = authentication_type("ansible_secret_key")
+    auth.nil? ? nil : auth.auth_key
+  end
+
+  def ansible_secret_key=(key)
+    auth = authentication_type("ansible_secret_key")
+    auth ||= AuthToken.new(
+      :name     => "Ansible Secret Key",
+      :resource => self,
+      :authtype => "ansible_secret_key"
+    )
+
+    auth.auth_key = key
+    auth.save!
+  end
+
+  def ansible_rabbitmq_password
+    auth = authentication_type("ansible_rabbitmq_auth")
+    auth.nil? ? nil : auth.password
+  end
+
+  def ansible_rabbitmq_password=(password)
+    auth = authentication_type("ansible_rabbitmq_auth")
+    auth ||= AuthUseridPassword.new(
+      :userid   => "tower",
+      :name     => "Ansible Rabbitmq Auth",
+      :resource => self,
+      :authtype => "ansible_rabbitmq_auth"
+    )
+
+    auth.password = password
+    auth.save!
+  end
+
+  def ansible_admin_password
+    auth = authentication_type("ansible_admin_password")
+    auth.nil? ? nil : auth.password
+  end
+
+  def ansible_admin_password=(password)
+    auth = authentication_type("ansible_admin_password")
+    auth ||= AuthUseridPassword.new(
+      :userid   => "admin",
+      :name     => "Ansible Admin Password",
+      :resource => self,
+      :authtype => "ansible_admin_password"
+    )
+
+    auth.password = password
+    auth.save!
+  end
```

Comment:
Yes, seems weird to have so many "ansible" specific references in `MiqDatabase`. It may seem like overkill for now, but maybe a new model for "appliance services" or something like that where ansible could be on of them. Just spitballing here...